src/bento/components/bento-soundcloud/1.0/test/standalone/test-bento-soundcloud.js

```diff
@@ -0,0 +1,238 @@
+      expect(iframe).to.not.be.null;
+
+      // Check iframe for correct scr URL
+      expect(iframe.src).to.equal(
+        buildURL('FF5500', undefined, '151584683', false, undefined)
+      );
+    });
+
+    it('renders secret token', async () => {
+      element = html`
+        <bento-soundcloud
+          data-trackid="243169232"
+          data-visual="false"
+          data-secret-token="c-af"
+        ></bento-soundcloud>
+      `;
+
+      // Add to Document
+      doc.body.appendChild(element);
+
+      // Wait till rendering is finished
+      await waitForRender();
+
+      // Extract iframe
+      const iframe = element.shadowRoot.querySelector('iframe');
+
+      // Make sure iframe is available
+      expect(iframe).to.not.be.null;
+
+      // Check iframe src for correct secret_token
+      expect(iframe.src).to.include(encodeURIComponent('?secret_token=c-af'));
+    });
+
+    it('renders without optional params', async () => {
+      element = html`
+        <bento-soundcloud data-trackid="243169232"></bento-soundcloud>
+      `;
+
+      // Add to Document
+      doc.body.appendChild(element);
+
+      // Wait till rendering is finished
+      await waitForRender();
+
+      // Extract iframe
+      const iframe = element.shadowRoot.querySelector('iframe');
+
+      // Make sure iframe is available
+      expect(iframe).to.not.be.null;
+
+      // Check iframe for correct scr URL
+      expect(iframe.src).not.to.include('&visual=true');
+      expect(iframe.src).not.to.include('&color=FF0000');
+    });
+
+    it('ignores color in visual mode', async () => {
+      element = html`
+        <bento-soundcloud
+          data-trackid="243169232"
+          data-visual="true"
+          data-color="00FF00"
+        >
+        </bento-soundcloud>
+      `;
+
+      // Add to Document
+      doc.body.appendChild(element);
+
+      // Wait till rendering is finished
+      await waitForRender();
+
+      // Extract iframe
+      const iframe = element.shadowRoot.querySelector('iframe');
+
+      // Make sure iframe is available
+      expect(iframe).to.not.be.null;
+
+      // Check iframe scr for correct parameters
+      expect(iframe.src).to.include('visual=true');
+      expect(iframe.src).not.to.include('color=00FF00');
+    });
+
+    it('renders data-trackid', async () => {
+      element = html`
+        <bento-soundcloud data-trackid="243169232"></bento-soundcloud>
+      `;
+
+      // Add to Document
+      doc.body.appendChild(element);
+
+      // Wait till rendering is finished
+      await waitForRender();
+
+      // Extract iframe
+      const iframe = element.shadowRoot.querySelector('iframe');
+
+      // Make sure iframe is available
+      expect(iframe).to.not.be.null;
+
+      /** Cannot test as userAssert not available for Preact Component */
+      // expect(iframe.src).to.be.rejectedWith(
+      //   /data-trackid or data-playlistid is required for <bento-soundcloud>/
+      // );
```

Comment:
Can this be deleted or else associated with a GitHub issue?

Reply:
this is directly copied from amp test, so i'll delete in both